tests/wsl/firstrun.pm

```diff
@@ -108,6 +108,15 @@ sub run {
         assert_screen 'local-user-credentials';
         enter_user_details([$realname, undef, $password, $password]);
         send_key 'alt-n';
+        # wsl-gui pattern installation
+        if (is_sle('>15-SP3')) {
+            assert_screen 'wsl-gui-pattern';
+            send_key 'alt-u' if (check_var('SLE_PRODUCT', 'SLED'));    # Install SLED instead of SLES
+            save_screenshot;    # Save screenshot to verify the checkbox was marked properly
+            send_key 'alt-i' if (is_sut_reg);    # wsl-gui requires registration
```

Comment:
I still think we should be able to test both options. We should be able to install `textmode` WSL (what we have now) and also `WSL_G`. The thing is that in order to install `WSL_G` I assume we need to have repositories enabled, therefore if we have no registration case, it won't be possible.

Reply:
I would also like to keep the logic of main simple and have a function for each screen as it is now.